### PR TITLE
test: fix tmpdir in test_cc_apk_configure

### PR DESCRIPTION
**Pull request by TheRealFalcon**
## Proposed Commit Message

```
test: fix tmpdir in test_cc_apk_configure

cc_apk_configure uses temp_utils.py, which has special logic to return
if the user is root that can't be retargeted using our current fixtures.
Fix it by setting that tmpdir in the test setup.
```

## Additional Context
Running tests in `tests/unittests/config/test_cc_apk_configure.py` as root will demonstrate the failure

## Merge type

- [x] Squash merge using "Proposed Commit Message"
- [ ] Rebase and merge unique commits. Requires commit messages per-commit each referencing the pull request number (#<PR_NUM>)

### CODE REVIEW

**Comment by TheRealFalcon** [comment]
cc @ani-sinha . For 23.4, it probably makes more sense to pin pytest to the version we were using when we released cloud-init 23.4 as you are doing, but this should fix the root cause of the problem you asked about in IRC.

File: tests/unittests/config/test_cc_apk_configure.py [hunk 1]
```diff
@@ -60,6 +60,7 @@ def setUp(self):
         self.name = "apk_configure"
         self.cloud = cloud.Cloud(None, self.paths, None, None, None)
         self.args = []
+        temp_utils._TMPDIR = self.new_root
```
Comment (holmanb) [inline tests/unittests/config/test_cc_apk_configure.py:63]: This sets a global for that module. Shouldn't we clean this up after tests run?

**Review by holmanb — COMMENTED** [review]
Nice fix, but don't we want to do cleanup of this as well?

File: tests/unittests/config/test_cc_apk_configure.py (hunk 1 above)
Comment (TheRealFalcon) [reply tests/unittests/config/test_cc_apk_configure.py:63]: Should be fixed now

**Review by holmanb — APPROVED** [review]
Looks good, thanks!

**Pull request merged**